### PR TITLE
Add isort and black

### PR DESCRIPTION
**Pull request by atugushev**
- adds `isort`
- adds `black`
- adds `checkqa` env to `tox`
- runs `checkqa` env on `python3`
- cleanup the code

### CODE REVIEW

File: piptools/resolver.py [hunk 1]
```diff
@@ -203,24 +228,32 @@ def _resolve_one_round(self):
...
+            log.debug("Removed dependencies in this round:")
+            for removed_dependency in sorted(
+                removed, key=lambda req: key_from_req(req.req)
+            ):
+                log.debug("  removing {}".format(removed_dependency))
+            log.debug("Unsafe dependencies in this round:")
+            for unsafe_dependency in sorted(
+                unsafe, key=lambda req: key_from_req(req.req)
+            ):
+                log.debug("  remembering unsafe {}".format(unsafe_dependency))
```
Comment (rpkilby) [inline piptools/resolver.py]: Is this block really more readable than the way it was before?

Comment (blueyed) [reply piptools/resolver.py]: Black is not about opinions.. ;)

I find this change OK (apart from the clutter that changing quotes brings here), but black can certainly make things less readable (https://github.com/ambv/black/issues/601).

Comment (rpkilby) [reply piptools/resolver.py]: The three lines aren't that much worse on their own - I'm referring to the entire block of code. Reading the previous section, the vertical flow visually reads better, while the new code doesn't read as well visually. 

> Black is not about opinions.. ;)

Isn't this exactly the point? Black has opinions so you don't have to?

Comment (blueyed) [reply piptools/resolver.py]: > Isn't this exactly the point? Black has opinions so you don't have to?

Yeah, that's what I've meant - "Black is not about (your) opinions".

I agree that the flow is worse now.

We could use "# fmt: off" / "# fmt: on" markers here.

File: setup.cfg [hunk 2]
```diff
@@ -7,3 +7,15 @@ license_file = LICENSE
 [tool:pytest]
 norecursedirs = .* build dist venv test_data piptools/_compat/* piptools/_vendored/*
 testpaths = tests piptools
+
+[flake8]
+# E501: line too long
+# W503: line break before binary operator
+ignore = E501, W503
```
[the remark this answers is not in the document]
Comment (atugushev) [reply setup.cfg]: In that case:
```bash
$ flake8 piptools tests/ | wc -l
      47
```

Should we split strings manually then? BTW, even black [ignore E501](https://github.com/ambv/black/blob/master/.flake8).

File: tox.ini [hunk 3]
```diff
@@ -33,9 +33,16 @@ setenv =
...
+[testenv:linters]
+basepython = python3
+deps =
+    flake8
+    black
+    isort
+commands =
+    black --check --diff .
+    isort --recursive --check-only --diff .
+    flake8 piptools tests
```
[the remark this answers is not in the document]
Comment (atugushev) [reply tox.ini]: Good point! Thanks.

File: piptools/resolver.py (hunk 1 above)
Comment (atugushev) [reply piptools/resolver.py]: IMO, this block looks okay (i've seen much worse). Anyways, PRs are welcome to fix "ugly" formattings.

File: .travis.yml [hunk 4]
```diff
@@ -51,8 +51,8 @@ jobs:
       python: "2.7"
 
     # QA checks.
-    - env: TOXENV=flake8
-      python: 2.7
+    - env: TOXENV=linters
```
[the remark this answers is not in the document]
Comment (atugushev) [reply .travis.yml]: Like it 👍

File: setup.cfg (hunk 2 above)
[the remark this answers is not in the document]
Comment (atugushev) [reply setup.cfg]: So, i've fixed long lines manually

**Comment by atugushev** [comment]
Travis got stuck :(

**Comment by atugushev** [comment]
Finally it's passed out!

**Comment by atugushev** [comment]
@blueyed are we okay with this PR?

**Pull request merged**